src/ImpactX.cpp

```diff
@@ -50,6 +50,28 @@ namespace impactx
     {
         BL_PROFILE("ImpactX::initGrids");
 
+        // n_cells has been set using temporary values earlier. We now know the true value of
+        // n_cells, so we recompute the Geometry objects for each level here *if* the user
+        // has set n_cells in the inputs file
+        {
+            amrex::Vector<int> n_cell(AMREX_SPACEDIM);
+            amrex::ParmParse pp_amr("amr");
+            pp_amr.queryarr("n_cell", n_cell);
+
+            amrex::IntVect lo(amrex::IntVect::TheZeroVector()), hi(n_cell);
+            hi -= amrex::IntVect::TheUnitVector();
+            amrex::Box index_domain(lo,hi);
+            for (int i = 0; i <= max_level; i++)
```

Comment:
Delayed MR init does not yet work with this. Let's revisit after #261 was merged :)